File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    int min = 100001, n = 0;
    
    if (arr.size() == 1) answer.push_back(-1);
    else {
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] < min) {
                min = arr[i];
                n = i;
            }
        }
        arr.erase(arr.begin() + n);
        answer = arr;
    }
    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[] {};
        List<int> lst = new List<int>(arr);
        List<int> sorted = new List<int>(arr);
        
        sorted.Sort();
        lst.Remove(sorted[0]);
        
        if (lst.Count == 0)
        {
            lst.Add(-1);
            answer = lst.ToArray();
        }
        else
        {
            answer = lst.ToArray();
            
        }
        return answer;
    }
}